chia/wallet/wallet.py

```diff
@@ -294,6 +291,41 @@ async def select_coins(self, amount, exclude: List[Coin] = None) -> Set[Coin]:
                 "Can't make this transaction at the moment. Waiting for the change from the previous transaction."
             )
 
+        # Use older coins first
+        valid_unspent.sort(reverse=True, key=lambda r: r.coin.amount)
+
+        # check for exact 1 to 1 coin match.
+        exact_match_coinrecord: Optional[WalletCoinRecord] = check_for_exact_match(valid_unspent, amount)
+        if exact_match_coinrecord:
+            used_coins.add(exact_match_coinrecord.coin)
+            self.log.debug(
+                f"Selected coin: {exact_match_coinrecord.coin.name()} "
+                f"at height {exact_match_coinrecord.confirmed_block_height}!"
+            )
+
+        # Check for an exact match with all of the coins smaller than the amount.
+        # If we have more, smaller coins than the amount we run the next algorithm.
+        if not exact_match_coinrecord:
+            smaller_coin_sum = 0  # coins smaller then target.
+            smaller_coins: Set[Coin] = set()
+            for coinrecord in valid_unspent:
+                if coinrecord.coin.amount < amount:
+                    smaller_coin_sum += coinrecord.coin.amount
+                    smaller_coins.add(coinrecord.coin)
+            if smaller_coin_sum == amount:
+                used_coins = smaller_coins
+            elif smaller_coin_sum < amount:
+                smallest_coin: Optional[WalletCoinRecord] = find_smallest_coin(valid_unspent, amount)
+                if smallest_coin:
+                    used_coins.add(smallest_coin.coin)
+                    self.log.debug(
+                        f"Selected coin: {smallest_coin.coin.name()} at height {smallest_coin.confirmed_block_height}!"
+                    )
+            elif smaller_coin_sum > amount:
+                best_coin_set, sum_best_coin_set = knapsack_coin_algorithm(smaller_coins, amount)
```

Comment:
I think this algorithm is guaranteed to succeed because pass 2 ends up adding all the remaining coins which add up to > amount